### DISCUSSION TITLE
Helm: changes in values.yaml file do not trigger reconciliation

### DISCUSSION THREAD

**Discussion by odombrovska** [fluxcd/flux2#3826] [Q&A]
I am using Flux as a GitOps extension for AKS cluster with Azure Storage Blob set as a source.

Here is my container:
![Screenshot_20230422_192612_Chrome.jpg](https://user-images.githubusercontent.com/77400214/233798346-561d534e-1ed7-4f05-9037-b85f8876cf02.jpg)

I can see stuff being pulled, e.g. when I apply a change in the release.yaml file, I can see that Flux sees the change and re-runs the configuration. I also noticed that every time I clean the container and push the files all over again, values.yaml file is being seen by Flux and correct values are applied, however, if I change values inside values.yaml, reconciliation doesn't get triggered and new values are not applied.

Here is my current release.yaml file: 

apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: gitops-poc
  namespace: flux-system
spec:
  interval: 1m
  releaseName: nginx
  chart: 
    spec:
      chart: ./
      valuesFiles:
       - values.yaml
      sourceRef:
        kind: Bucket
        name: helm
  targetNamespace: gitops-poc

I tried using valuesFrom with chartFileRef, however, I think (?) this is no longer supported in helm.toolkit.fluxcd.io/v2beta1

How can I fix this? My values file is a .yaml file, an ordinary Helm values file. Should I create ConfigMaps for them and then reference via valuesFrom? What is the recommended approach here?

Thanks!

**Comment by kingdonb** [comment]
If you use the `Revision` `reconcileStrategy` for your HelmRelease's `spec.chart.spec`, it should solve this problem.

HelmReleases are not recommended to be used with GitRepository or Bucket sources except in special cases. It is much better to use a HelmRepository source, or better yet: a HelmRepository source with `type: oci` that is much more efficient than the legacy kind.

Please check out this note in the HelmReleases guide about Cloud Storage as a source:

https://fluxcd.io/flux/guides/helmreleases/#cloud-storage
